### ISSUE TITLE
[request] limit window ratio when resizing

### ISSUE THREAD

**Issue by makedir** [moonlight-stream/moonlight-qt#1317]
Please implement an option (or make it default), that if you resize the window, it limits the size to same ratio of the stream and wont stretch.

**Comment by cgutman** [comment]
> Please implement an option (or make it default), that if you resize the window (when you use window mode), it limits the size to same ratio of the stream and wont stretch. Also limit the size so the divider is even and the picture wont be blurry if not.

I'm not sure I understand what you mean. If you mean to scale the video preserving the aspect ratio, Moonlight already behaves like that. All scaling on the Moonlight side preserves aspect ratio, hence the pillarboxing/letterboxing you will see if you stretch the content.

> Also as a 2nd request, if possible, make an option, to remove the border / window decoration. So it is a borderless window but you can resize it with mouse drag and drop.

> As a 3rd request, and as an alternative to 2nd request, would be nice, if you could optional give widh and height of borderless window mode not full screen but give a downscale size. And you can move the window aroundw ith the mouse.

I'm not sure about these. I don't think a borderless window can be resized or repositioned like that because the window decorations are what the user interacts with to move and resize the window. Moonlight would have to support its own resize/move functionality internally, which would then have to be deconflicted with the normal mouse interactions that users expect to have with their host PC via a Moonlight session.

I guess I could provide a command-line option and/or environment variable override for the default window sizing logic if that helps.

**Comment by makedir** [comment]
No, I meant the window of moonlight not the video scaling properly with same scale factor, see:

![image](https://github.com/moonlight-stream/moonlight-qt/assets/6721083/209d96ff-7454-48d8-ad3d-bf4fd8c95cd5)

This is not wanted when using window mode. Please implement a logic, that if you resize Moonlight, the window of Moonlight cant be resized into other ratios. For example if video is 16:9 the resizing of Moonlight always is fixed to 16:9 factor. Do you know the video player mpc? it has an option like that:

![image](https://github.com/moonlight-stream/moonlight-qt/assets/6721083/c2ae0115-50c3-4a57-8b67-b2a979cb4cd7)

Please implement the same into Moonlight, it would be really handy.

For the 2nd one, yes would be possible MPC behaves like it too and you also can move around the window without border or gui around, so a window looks like this:

![image](https://github.com/moonlight-stream/moonlight-qt/assets/6721083/5ea2aea1-3b99-4add-a684-1babcf17d330)

You can move the mouse cursor ro the corner and resize and with above option it scales properly with same aspect ratio.

**Comment by chopin1998** [comment]
that's true, need keep window aspect, and/or add 1:1 options

**Comment by TheRedfoox** [comment]
Has there been movement on the subject? I'm really looking for a way not to have the window decoration under macOS. When I want to stream my PC I would like not to have the decoration visible. Moving the window is not a problem because I have a layout software that automatically places the window in a specific dimension and place.
